### DISCUSSION TITLE
v0.8.4.2 - Hotfix + Migration Script

### DISCUSSION THREAD

**Discussion by majora2007** [Kareadita/Kavita#3424] [Announcements]
A few small things slipped past. 

I want to take this space to first apologize to our users that manually tag their metadata in Kavita. Post release has been an eye opener to the different ways uses of Kavita and a lesson learned to work harder to preserve data between releases. I always try, but sometimes big lift and shifts like this are needed and aren't easy to preserve data. I will take it on myself to strive for excellence and do what is feasible to preserve data. I don't foresee many major refactors like this happening in the near future. 

On more positive news, MetronInfo.xml has become stabilized. This is a step towards advancing the metadata format for Comics and Manga (and potentially books in the future). I have worked closely with Brian (the creator) to include Manga into the format. Kavita is dedicated towards supporting the format, that work is planned for next year. Read up more about it [here](https://metron-project.github.io/blog/metroninfo-v1).

If you have a lot of metadata that was tagged just via the UI, you can try this [script](https://pastebin.com/RPmfSSYq) a user wrote to migrate the data. Grab a backup from v0.8.3 and run this against it. You need to update `oldDb` and `newDb`. I cannot offer support for this script. 

I did convert this to python via ChatGPT and added instructions, it will be much easier but I did not test it. You can find the script [here](https://pastebin.com/BR0GxaXB).

# Fixed
- Fixed: Send to on mobile was still broken
- Fixed: Changed On Deck to use a workaround as a temp fix
- Fixed: Fixed a bug where on non-English locales parsing numbers would throw an exception and cause files to not be ingested. 
- Fixed: Bulk library delete wasn't hooked up
- Fixed: Fixed a bug where updating chapter metadata with a new person, the normalized name was saved
- Fixed: Potentially fixed a bug where adding multiple people on a series via UI will throw an exception (Was not able to reproduce, but added extra safety logic)

<hr /><em>This discussion was created from the release <a href='https://github.com/Kareadita/Kavita/releases/tag/v0.8.4.2'>v0.8.4.2 - Hotfix + Migration Script</a>.</em>

**Comment by projector22** [comment]
Really appreciate the time and effort bud

**Comment by maximo1984** [comment]
I understand bugs sneak through. Appreciate the ongoing work and still here to support.

**Comment by infinitel8p** [comment]
Thank you for the quick fix!

**Comment by MVConsorte** [comment]
Hey guys, I would like to know if any of the versions would run on x86 architecture Linux?

**Reply by DieselTech** [reply]
👀 Yeah that is an oversight that hasn't been brought up before. Let me get back to you.

**Reply by MVConsorte** [reply]
Thanks for letting me know and for your attention. Basically, I am setting up an old test laptop (Antix Linux for 32 bits) and the idea was to get Kavita running on it. Anyway, i'll be waiting for a response.

**Reply by DieselTech** [reply]
> Thanks for letting me know and for your attention. Basically, I am setting up an old test laptop (Antix Linux for 32 bits) and the idea was to get Kavita running on it. Anyway, i'll be waiting for a response.

I looked into it more and I'm sorry to report that Microsoft has discontinued x86 .NET support all the way back in .NET 5. I incorrectly thought it was an oversight on our part since we have ARM 32-bit, but traditional 32-bit on PC just doesn't have the user base anymore for them to keep supporting it since we have had 64-bit CPU's since 2004.
